lib/goliath/request.rb

```diff
@@ -237,11 +237,6 @@ def server_exception(e)
       headers['Content-Length'] = body.bytesize.to_s
       @env[:terminate_connection] = true
       post_process([status, headers, body])
-
-      # Mark the request as complete to force a flush on the response.
-      # Note: #on_body and #response hooks may still fire if the data
-      # is already in the parser buffer.
-      succeed
```

Comment:
Why are we removing this succeed call?

Reply:
Let's consider the following scenario:

* A request comes in and it's successfully handled
* We start streaming the response for that request to the user
* In the meanwhile another request comes using the same TCP connection
* This request errors, and `#server_exception` is called
* If `#server_exception` marks the request as succeeded, the error response is immediately written to the socket
* Since the response for the first request hasn't been written yet, we get interleaved responses

We remove the `succeed` code so that `Goliath::Connection` is now the only one deciding whether the request is successful, so that we handle multiple requests over the same connection in one place. `Goliath::Connection` waits for the current request to finish before marking any pending ones that came during as completed.

Reply:
Ok, I think that makes sense — thanks for the detailed explainer.